src/rule.cpp

```diff
@@ -15,52 +15,28 @@
 namespace ddwaf
 {
 
-bool condition::matchWithTransformer(const ddwaf_object* baseInput, MatchGatherer& gatherer, bool onKey, bool readOnlyArg) const
+bool condition::matchWithTransformer(const ddwaf_object* baseInput, MatchGatherer& gatherer) const
 {
     const bool hasTransformation        = !transformation.empty();
-    const bool canRunTransformation     = onKey || (baseInput->type == DDWAF_OBJ_STRING);
+    const bool canRunTransformation     = baseInput->type == DDWAF_OBJ_STRING;
     bool transformationWillChangeString = false;
 
     if (hasTransformation && canRunTransformation)
     {
         // This codepath is shared with the mutable path. The structure can't be const :/
-        if (onKey && baseInput != NULL)
-        {
-            ddwaf_object fakeArg;
-            ddwaf_object_stringl_nc(&fakeArg, baseInput->parameterName, baseInput->parameterNameLength);
-            transformationWillChangeString = PWTransformer::doesNeedTransform(transformation, &fakeArg);
-        }
-        else
-        {
-            transformationWillChangeString = PWTransformer::doesNeedTransform(transformation, (ddwaf_object*) baseInput);
-        }
+        transformationWillChangeString = PWTransformer::doesNeedTransform(transformation,
+            const_cast<ddwaf_object *>(baseInput));
     }
 
     //If we don't have transformation to perform, or if they're irrelevant, no need to waste time copying and allocating data
     if (!hasTransformation || !canRunTransformation || !transformationWillChangeString)
     {
-        if (onKey)
-            return processor->doesMatchKey(baseInput, gatherer);
         return processor->doesMatch(baseInput, gatherer);
     }
 
     ddwaf_object copyInput;
-    if (readOnlyArg)
-    {
-        // Copy the input. If we're running on the key, we copy it in the value as it's functionnaly equivalent
-        if (onKey)
-        {
-            ddwaf_object_stringl(&copyInput, (const char*) baseInput->parameterName, baseInput->parameterNameLength);
-        }
-        else
-        {
-            ddwaf_object_stringl(&copyInput, (const char*) baseInput->stringValue, baseInput->nbEntries);
-        }
-    }
-    else
-    {
-        copyInput = *baseInput;
-    }
+    // Copy the input. If we're running on the key, we copy it in the value as it's functionnaly equivalent
```

Comment:
The comment is stale. The key iterator provides only string ddwaf_objects already.